subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/report/AbstractTablePageGenerator.java

```diff
@@ -262,4 +264,14 @@ private void renderVersionHeader(String version) {
             }
             // @formatter:on
     }
+
+    private Optional<String> determineBaseline() {
+        return executionDataProvider.getReportScenarios().stream()
+            .filter(scenario -> !scenario.getCrossBuild())
+            .findFirst()
+            .flatMap(scenario -> scenario.getHistoryExecutions().isEmpty()
+                ? Optional.empty()
+                : Optional.of(scenario.getHistoryExecutions().get(0).getBaseVersion().getName())
+            );
```

Comment:
For an `Optional`, you can actually `map` to `null` and then you don't need to use `flatMap`. Another option would be to use `filter` followed by `map`, which would make the code easier to understand I think. Something like this:

```suggestion
            .filter(scenario -> !scenario.getHistoryExecutions().isEmpty())
            .map(scenario -> scenario.getHistoryExecutions().get(0).getBaseVersion().getName());
```

Reply:
👍